### ISSUE TITLE
Update LICENSE.md after approval

### ISSUE THREAD

**Issue by steven-bellock** [DMTF/libspdm#1481]
https://github.com/DMTF/libspdm/blob/main/LICENSE.md contains http as well, but that comes from https://www.dmtf.org/sites/default/files/DMTF_Software_Submissions-2014.03_0.pdf We'll need to update the document before we can change the license file.

_Originally posted by @steven-bellock in https://github.com/DMTF/libspdm/issues/1462#issuecomment-1335629926_

**Comment by steven-bellock** [comment]
`http` was changed to `https` so I will close this issue. I will create a separate issue if we need to remove "Distributed Management Task Force" from the license file.
